src/main/java/no/nav/tag/kontakt/oss/vault/HikariCPVaultUtil.java

```diff
@@ -0,0 +1,102 @@
+package no.nav.tag.kontakt.oss.vault;
+
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultException;
+import com.bettercloud.vault.response.LogicalResponse;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.TimerTask;
+
+public final class HikariCPVaultUtil {
+
+    private static final Logger logger = LoggerFactory.getLogger(HikariCPVaultUtil.class);
+
+    private HikariDataSource ds = null;
+    private final HikariConfig hikariConfig;
+    private final Vault vault;
+    private final String mountPath;
+    private final String role;
+
+    private HikariCPVaultUtil(final HikariConfig config, final Vault vault, final String mountPath, final String role) {
+        this.hikariConfig = config;
+        this.vault = vault;
+        this.mountPath = mountPath;
+        this.role = role;
+    }
+
+    private void setDs(final HikariDataSource ds) {
+        this.ds = ds;
+    }
+
+    public static HikariDataSource createHikariDataSourceWithVaultIntegration(final HikariConfig config, final String mountPath, final String role) throws VaultError {
+        final VaultUtil instance = VaultUtil.getInstance();
+        final Vault vault = instance.getClient();
+
+        final HikariCPVaultUtil hikariCPVaultUtil = new HikariCPVaultUtil(config, vault, mountPath, role);
+
```

Comment:
Synes det er litt uklart ansvarsforhold mellom `DatabaseConfig` og `HikariCPVaultUtil`. `DatabaseConfig` setter opp deler av HikariConfig, men overlater til `HikariCPVaultUtil` å faktisk opprette HikariDataSource-instansen i tillegg til å stadig oppdatere brukernavn og passord. Tror det kunne vært litt ryddigere om den VaultUtil-en bare tar seg av det å hente nye credentials, mens DatabaseConfig gjør resten av det som har med datasource å gjøre, men er ikke helt sikker på hva som vil være en god struktur for det :)